Neos.Media/Classes/Command/MediaCommandController.php

```diff
@@ -164,35 +168,57 @@ public function removeUnusedCommand(string $assetSource = '', bool $quiet = fals
             !$quiet && $this->outputLine('<b>Searching for unused assets of asset source "%s":</b>', [$filterByAssetSourceIdentifier]);
         }
 
+        $assetTagsMatchFilterTags = function (Collection $assetTags, string $filterTags): bool {
+            $filterTagValues = Arrays::trimExplode(',', $filterTags);
+            $assetTagValues = [];
+            foreach ($assetTags as $tag) {
+                /** @var Tag $tag */
+                $assetTagValues[] = $tag->getLabel();
+            }
+            return count(array_intersect($filterTagValues, $assetTagValues)) > 0;
+        };
+
         !$quiet && $this->output->progressStart($assetCount);
 
         foreach ($this->assetRepository->iterate($iterator) as $asset) {
-            if ($asset instanceof Asset && $asset->getUsageCount() === 0) {
-                if (!$asset instanceof AssetSourceAwareInterface) {
-                    continue;
-                }
-                if ($filterByAssetSourceIdentifier !== '' && $asset->getAssetSourceIdentifier() !== $filterByAssetSourceIdentifier) {
-                    continue;
-                }
-
-                $fileSize = str_pad(Files::bytesToSizeString($asset->getResource()->getFileSize()), 9, ' ', STR_PAD_LEFT);
-
-                $unusedAssets[] = $asset;
-                $tableRowsByAssetSource[$asset->getAssetSourceIdentifier()][] = [
-                    $asset->getIdentifier(),
-                    $asset->getResource()->getFilename(),
-                    $fileSize
-                ];
-                $unusedAssetCount++;
-                $unusedAssetsTotalSize += $asset->getResource()->getFileSize();
-            }
             !$quiet && $this->output->progressAdvance(1);
+
+            if ($unusedAssetCount === $limit) {
+                break;
+            }
+
+            if (!($asset instanceof Asset)) {
+                continue;
+            }
+            if (!$asset instanceof AssetSourceAwareInterface) {
+                continue;
+            }
+            if ($filterByAssetSourceIdentifier !== '' && $asset->getAssetSourceIdentifier() !== $filterByAssetSourceIdentifier) {
+                continue;
+            }
+            if (trim($onlyTags) && !($assetTagsMatchFilterTags($asset->getTags(), $onlyTags))) {
```

Comment:
`trim($onlyTags)` is a weak comparison. Why not just `$onlyTags !== ''` or make it `null` by default?